diff --git a/.travis.yml b/.travis.yml
index 5ab5d03..40c43d0 100644
--- a/.travis.yml
+++ b/.travis.yml
@@ -12,3 +12,5 @@ before_script:
 script:
   - cargo clippy -- -D clippy::pedantic
   - cargo test --all
+
+# for test the only add diff of .rs files
diff --git a/src/git.rs b/src/git.rs
index 5e6c31e..1816c80 100644
--- a/src/git.rs
+++ b/src/git.rs
@@ -91,8 +91,13 @@ impl Parser {
// File added or edited
// +++ b/Cargo.lock
if l.starts_with("+++") {
-// TODO: do something less ugly with the bounds and indexing
-file_name = l[l.find('/').unwrap() + 1..].into();
+// Only add .rs file to the diff
+if l.ends_with(".rs") {
+    // TODO: do something less ugly with the bounds and indexing
+    file_name = l[l.find('/').unwrap() + 1..].into();
+} else {
+    continue;
+}
}
